utils/build-script-impl

```diff
@@ -1475,8 +1554,28 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
         unset skip_build
         source_dir_var="$(toupper ${product})_SOURCE_DIR"
         source_dir=${!source_dir_var}
-        build_dir=$(build_directory ${deployment_target} ${product})
+        build_dir=$(build_directory ${host} ${product})
         build_targets=(all)
+
+        if [[ $(is_cross_tools_host ${host}) ]] ; then
+
+            # Cross-compiled hosts are always compiled after the LOCAL_HOST,
+            # and are built with the freshly-baked tools we just made.
```

Comment:
There was some talk about cross-compiling before and I believe that was what was suggested. It doesn't have to be; but it's probably better as the default. I could add a flag to bypass it.